crates/amaru-stores/src/rocksdb/mod.rs

```diff
@@ -160,44 +163,56 @@ impl RocksDB {
     fn transaction_ended(&self) {
         self.ongoing_transaction.set(false);
     }
+}
 
-    fn get<T: for<'d> cbor::decode::Decode<'d, ()>>(
-        &self,
-        key: &str,
-    ) -> Result<Option<T>, StoreError> {
-        self.db
-            .get(key)
-            .map_err(|err| StoreError::Internal(err.into()))?
-            .map(|bytes| cbor::decode(&bytes))
-            .transpose()
-            .map_err(StoreError::Undecodable)
-    }
+fn get<T: for<'d> cbor::decode::Decode<'d, ()>>(
+    db: &OptimisticTransactionDB,
+    key: &str,
+) -> Result<Option<T>, StoreError> {
+    db.get(key)
+        .map_err(|err| StoreError::Internal(err.into()))?
+        .map(|bytes| cbor::decode(&bytes))
+        .transpose()
+        .map_err(StoreError::Undecodable)
 }
 
 #[allow(clippy::panic)]
 #[allow(clippy::unwrap_used)]
 fn iter<'a, K: Clone + for<'d> cbor::Decode<'d, ()>, V: Clone + for<'d> cbor::Decode<'d, ()>>(
     db: &OptimisticTransactionDB,
     prefix: [u8; PREFIX_LEN],
+    direction: Direction,
 ) -> Result<impl Iterator<Item = (K, V)> + use<'_, K, V>, StoreError> {
-    Ok(db.prefix_iterator(prefix).map(|e| {
-        let (key, value) = e.unwrap();
-        let decoded_key = cbor::decode(&key[PREFIX_LEN..])
-            .unwrap_or_else(|e| panic!("unable to decode key ({}): {e:?}", hex::encode(&key)));
-        let decoded_value = cbor::decode(&value).unwrap_or_else(|e| {
-            panic!(
-                "unable to decode value ({}) for key ({}): {e:?}",
-                hex::encode(&key),
-                hex::encode(&value)
-            )
-        });
-        (decoded_key, decoded_value)
-    }))
+    let mut opts = ReadOptions::default();
+    opts.set_prefix_same_as_start(true);
+    Ok(db
+        .iterator_opt(IteratorMode::From(prefix.as_ref(), direction), opts)
```

Comment:
Any reason for the change 🤔 ?

Reply:
This is in preparation for the implementation of `get_protocol_parameters_for `